static/app/views/issueList/header.tsx

```diff
@@ -130,11 +132,11 @@ function IssueListHeader({
         </ButtonBar>
       </Layout.HeaderActions>
       <StyledGlobalEventProcessingAlert projects={selectedProjects} />
-      <StyledTabs selectedKey={query} onSelectionChange={() => {}}>
+      <StyledTabs value={tabValues.has(query) ? query : CUSTOM_TAB_VALUE}>
```

Comment:
When `query` doesn't match any of the tab items within the bar (e.g. user entered a custom query), `<Tabs />` automatically falls back to a default, valid value, which in this case is just the first item's value, and `onChange` is called once. Here we work around the issue by overwriting `onSelectionChange` (the internal prop that triggers `onChange`) with an empty function. `onSelectionChange` will be deprecated soon (in favor of `onChange`, https://github.com/getsentry/sentry/pull/51755), so we can't use this workaround anymore.